Gems/ROS2/Code/Include/ROS2/Frame/ROS2FrameEditorComponent.h

```diff
@@ -0,0 +1,95 @@
+
+        ROS2FrameEditorComponent() = default;
+        ~ROS2FrameEditorComponent() = default;
+        //! Initialize to a specific frame id
+        ROS2FrameEditorComponent(const AZStd::string& frameId);
+
+        //////////////////////////////////////////////////////////////////////////
+        // Component overrides
+        void Activate() override;
+        void Deactivate() override;
+        //////////////////////////////////////////////////////////////////////////
+
+        static void Reflect(AZ::ReflectContext* context);
+        static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
+        static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+
+        //! Get a frame id, which is needed for any ROS2 message with a Header
+        //! @return Frame id which includes the namespace, ready to send in a ROS2 message
+        AZStd::string GetFrameID() const override;
+
+        //! Set a above-mentioned frame id
+        void SetFrameID(const AZStd::string& frameId);
+
+        //! Get the joint name including the namespace
+        //! @note Supplementary metadata for Joint components, necessary in some cases for joints addressed by name in ROS 2
+        //! @return The namespaced joint name, ready to send in a ROS2 message
+        AZ::Name GetJointName() const override;
+
+        //! Set the joint name
+        //! @note May be populated during URDF import or set by the user in the Editor view
+        //! @param jointNameString does not include the namespace. The namespace prefix is added automatically.
+        void SetJointName(const AZStd::string& jointNameString);
+
+        //! Get a namespace, which should be used for any publisher or subscriber in the same entity.
+        //! @return A complete namespace (including parent namespaces)
+        AZStd::string GetNamespace() const override;
+
+        //! Update the parents namespace
+        //! @param parentsNamespace
+        void OnNamespaceChange(AZStd::string parentsNamespace) override;
+
+        //! Updates the namespace and namespace strategy of the underlying namespace configuration
+        //! @param ns Namespace to set.
+        //! @param strategy Namespace strategy to use.
+        void UpdateNamespaceConfiguration(const AZStd::string& ns, NamespaceConfiguration::NamespaceStrategy strategy);
+
+        //! Global frame name in ros2 ecosystem.
+        //! @return The name of the global frame with namespace attached. It is typically "odom", "map", "world".
+        AZStd::string GetGlobalFrameName() const override;
+
+        void BuildGameEntity(AZ::Entity* gameEntity) override;
+
+    private:
+        bool IsTopLevel() const; //!< True if this entity does not have a parent entity with ROS2.
+
+        AZ::Crc32 OnConfigurationChange();
```

Comment:
It should be override, shouldn't it? Add base class name (ROS2FrameSystemRequests)